### ISSUE TITLE
Use WebDriverExtensions with Serenity framework

### ISSUE THREAD

**Issue by matthew6870** [webdriverextensions/webdriverextensions#130]
Hello.

We have JUnit project using Serenity framework with custom Selenium webdriver for testing of mobile applications.

We tried to integrate WebDriverExtensions to be able to use WebComponents within PageObjects.

- we use kotlin
- webdriverextensions version 3.11.0
- serenity 2.0.41
- we can't use WebDriverRunner, since we are using Serenity runner
- our PageObjects are using serenity's FindBy annotations

Our custom MobileWebDriver is initialized using DriverSource interface from Serenity library

**We have created simple WebComponent and inserted it to PageObject.**

Error:
> net.thucydides.core.pages.WrongPageError: The page object class xxx.android.pages.StartPage looks dodgy:
Failed to instantiate page (java.lang.IllegalArgumentException: Can not set xxx.android.components.Button field xxx.android.pages.StartPage.btnLoginComponent to com.sun.proxy.$Proxy51)

Our custom WebComponent is issue.

```
class Button : WebComponent(){
    @FindBy(xpath = "//*[@id='button_text_title']")
    private lateinit var btnTextTitle: WebElementFacade

    @FindBy(xpath = "//*[@id='button_icon']")
    private lateinit var btnIcon: WebElementFacade
}
```

**After we tried to call in PageObject's constructor**
`    PageFactory.initElements(WebDriverExtensionFieldDecorator(driver), page);
`

Error:
> net.thucydides.core.pages.WrongPageError: The page object class xxx.android.pages.virtualcard.VirtualCardPage looks dodgy:
Failed to instantiate page (java.lang.IllegalArgumentException: Can not set net.serenitybdd.core.pages.WebElementFacade field xxx.android.pages.xxx.btnTransfer to com.sun.proxy.$Proxy48)

WebElementFacade field is issue now.

Thanks for your help

**Comment by matthew6870** [comment]
@andidev any update?

**Comment by andidev** [comment]
I think this may be the problemn

```
our PageObjects are using serenity's FindBy annotations 
```

Not sure what the serenity framework does but my guesses are that WebDriverExtensionFieldDecorator does not recognize the find by annotations. How does serenity frameworks decorator look like? Another problem may also be WebElementFacade. Seems to be a serenity specific thing. Is it an interface? Does a standard WebElement implement that interface?

I don't think it is possible to use the serenity framework without creating a custom WebDriverExtensionFieldDecorator.

**Comment by matthew6870** [comment]
Yes, WebElementFacade is serenity specific interface. WebElementFacade class contains convenient API for dealing with web elements, providing some commonly-used extra features that are not provided out-of-the-box by the WebDriver API.  WebElementFacade implements WebElement interface.

We do not use any decorators (or we don't know about any).

Do you plan to support Serenity framework in future?

**Comment by andidev** [comment]
Sorry, no plan for that. I have very limited time right now. But feel free to create a new doctorater and I'll add it to webdriverextensions. Should not be to mutch work.
